src/implementation/MassaWalletProviders.ts

```diff
@@ -0,0 +1,120 @@
+import { uid } from 'uid';
+import { ICustomEventMessageResponse } from '../interfaces/ICustomEventMessageResponse';
+import { ICustomEventMessageRequest } from '../interfaces/ICustomEventMessageRequest';
+import { AvailableCommands } from './Commands';
+import { IRegisterEvent } from '../interfaces';
+
+const MASSA_WINDOW_OBJECT = 'massaWalletProvider';
+
+type CallbackFunctionVariadicAnyReturn = (
+  error: Error | null,
+  result: any,
+) => any;
+
+// =========================================================
+
+export class MassaWalletProviders {
+  private static instance: MassaWalletProviders;
+  private registeredProviders: { [key: string]: string } = {};
+  private pendingRequests: Map<string, CallbackFunctionVariadicAnyReturn>;
+
+  public static init(): MassaWalletProviders {
```

Comment:
I don't understand what this function is doing. If you want to instantiate MassaWalletprovider, why not doing so and export it?

Reply:
removed, wanted a singleton pattern, but your idea is better to just initialzie it. Done